Implement Camera Management Dashboard with analytics and camera confi…

Shivansh1357 opened the pull request:
…guration features

- Added new pages for camera management, analytics, and demographics configuration.
- Integrated React Query for data fetching and state management.
- Created reusable UI components for buttons, cards, and inputs.
- Established API client for handling requests to the backend.
- Updated package dependencies in package.json and package-lock.json.
- Added setup instructions in SETUP.md for easy onboarding.

Shivansh1357 committed:
Implement Camera Management Dashboard with analytics and camera configuration features

- Added new pages for camera management, analytics, and demographics configuration.
- Integrated React Query for data fetching and state management.
- Created reusable UI components for buttons, cards, and inputs.
- Established API client for handling requests to the backend.
- Updated package dependencies in package.json and package-lock.json.
- Added setup instructions in SETUP.md for easy onboarding.

shafay07 commented:
@Shivansh1357 
<details>

<summary>
Build logs
</summary>

```
2025-Jun-16 16:22:54.637818
#10 [stage-0  6/11] COPY . /app/.
2025-Jun-16 16:22:54.637818
#10 DONE 0.1s
2025-Jun-16 16:22:54.637818
2025-Jun-16 16:22:54.637818
#11 [stage-0  7/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-/root/npm,target=/root/.npm npm ci
2025-Jun-16 16:22:54.637818
#11 0.657 npm warn config production Use `--omit=dev` instead.
2025-Jun-16 16:22:54.637818
#11 20.13
2025-Jun-16 16:22:54.637818
#11 20.13 added 389 packages, and audited 390 packages in 20s
2025-Jun-16 16:22:54.637818
#11 20.13
2025-Jun-16 16:22:54.637818
#11 20.13 145 packages are looking for funding
2025-Jun-16 16:22:54.637818
#11 20.13   run `npm fund` for details
2025-Jun-16 16:22:54.637818
#11 20.13
2025-Jun-16 16:22:54.637818
#11 20.13 1 low severity vulnerability
2025-Jun-16 16:22:54.637818
#11 20.13
2025-Jun-16 16:22:54.637818
#11 20.13 To address all issues, run:
2025-Jun-16 16:22:54.637818
#11 20.13   npm audit fix
2025-Jun-16 16:22:54.637818
#11 20.13
2025-Jun-16 16:22:54.637818
#11 20.13 Run `npm audit` for details.
2025-Jun-16 16:22:54.637818
#11 DONE 21.0s
2025-Jun-16 16:22:54.637818
2025-Jun-16 16:22:54.637818
#12 [stage-0  8/11] COPY . /app/.
2025-Jun-16 16:22:54.637818
#12 DONE 0.3s
2025-Jun-16 16:22:54.637818
2025-Jun-16 16:22:54.637818
#13 [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build
2025-Jun-16 16:22:54.637818
#13 0.580 npm warn config production Use `--omit=dev` instead.
2025-Jun-16 16:22:54.637818
#13 0.614
2025-Jun-16 16:22:54.637818
#13 0.614 > my-app@0.1.0 build
2025-Jun-16 16:22:54.637818
#13 0.614 > next build
2025-Jun-16 16:22:54.637818
#13 0.614
2025-Jun-16 16:22:54.637818
#13 1.924    ▲ Next.js 15.3.3
2025-Jun-16 16:22:54.637818
#13 1.925
2025-Jun-16 16:22:54.637818
#13 1.989    Creating an optimized production build ...
2025-Jun-16 16:22:54.637818
#13 18.08  ✓ Compiled successfully in 11.0s
2025-Jun-16 16:22:54.637818
#13 18.08    Linting and checking validity of types ...
2025-Jun-16 16:22:54.637818
#13 22.23
2025-Jun-16 16:22:54.637818
#13 22.23 Failed to compile.
2025-Jun-16 16:22:54.637818
#13 22.23
2025-Jun-16 16:22:54.637818
#13 22.23 ./src/app/analytics/page.tsx
2025-Jun-16 16:22:54.637818
#13 22.23 18:3  Error: 'Zap' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-16 16:22:54.637818
#13 22.23
2025-Jun-16 16:22:54.637818
#13 22.23 ./src/app/page.tsx
2025-Jun-16 16:22:54.637818
#13 22.23 11:3  Error: 'Globe' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-16 16:22:54.637818
#13 22.23
2025-Jun-16 16:22:54.637818
#13 22.23 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-16 16:22:54.637818
#13 ERROR: process "/bin/bash -ol pipefail -c npm run build" did not complete successfully: exit code: 1
2025-Jun-16 16:22:54.637818
------
2025-Jun-16 16:22:54.637818
> [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build:
2025-Jun-16 16:22:54.637818
22.23
2025-Jun-16 16:22:54.637818
22.23 Failed to compile.
2025-Jun-16 16:22:54.637818
22.23
2025-Jun-16 16:22:54.637818
22.23 ./src/app/analytics/page.tsx
2025-Jun-16 16:22:54.637818
22.23 18:3  Error: 'Zap' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-16 16:22:54.637818
22.23
2025-Jun-16 16:22:54.637818
22.23 ./src/app/page.tsx
2025-Jun-16 16:22:54.637818
22.23 11:3  Error: 'Globe' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-16 16:22:54.637818
22.23
2025-Jun-16 16:22:54.637818
22.23 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-16 16:22:54.637818
------
2025-Jun-16 16:22:54.637818
Dockerfile:24
2025-Jun-16 16:22:54.637818
--------------------
2025-Jun-16 16:22:54.637818
22 |     # build phase
2025-Jun-16 16:22:54.637818
23 |     COPY . /app/.
2025-Jun-16 16:22:54.637818
24 | >>> RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build
2025-Jun-16 16:22:54.637818
25 |
2025-Jun-16 16:22:54.637818
26 |
2025-Jun-16 16:22:54.637818
--------------------
2025-Jun-16 16:22:54.637818
ERROR: failed to solve: process "/bin/bash -ol pipefail -c npm run build" did not complete successfully: exit code: 1
2025-Jun-16 16:22:54.637818
exit status 1
```
</details>

Shivansh1357 committed:
Fix build errors: Remove unused imports - Build now passes successfully with no ESLint errors

Shivansh1357 commented:
> @Shivansh1357
> 
> Build logs

Resolved!  

Merge please

Shivansh1357 committed:
Refactor analytics and camera management components

- Updated analytics page to utilize new demographics data structure, improving data handling and display.
- Enhanced camera management page with status indicators and snapshot previews for better user experience.
- Implemented delete functionality for demographics configurations, including confirmation dialog.
- Improved API integration for demographics results and camera details, ensuring accurate data retrieval.
- Adjusted UI components for better responsiveness and visual consistency.

Shivansh1357 commented:
> @Shivansh1357
> 
> Build logs

Hey, will you please merge and review again ?

shafay07 commented:
@Shivansh1357 the latest build has failed again please refer to the logs below

<details>
<summary>
Build logs
</summary>

```
2025-Jun-17 11:28:35.932882
#12 [stage-0  8/11] COPY . /app/.
2025-Jun-17 11:28:35.932882
#12 DONE 0.3s
2025-Jun-17 11:28:35.932882
2025-Jun-17 11:28:35.932882
#13 [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build
2025-Jun-17 11:28:35.932882
#13 0.673 npm warn config production Use `--omit=dev` instead.
2025-Jun-17 11:28:35.932882
#13 0.710
2025-Jun-17 11:28:35.932882
#13 0.710 > my-app@0.1.0 build
2025-Jun-17 11:28:35.932882
#13 0.710 > next build
2025-Jun-17 11:28:35.932882
#13 0.710
2025-Jun-17 11:28:35.932882
#13 2.032 Attention: Next.js now collects completely anonymous telemetry regarding usage.
2025-Jun-17 11:28:35.932882
#13 2.033 This information is used to shape Next.js' roadmap and prioritize features.
2025-Jun-17 11:28:35.932882
#13 2.033 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
2025-Jun-17 11:28:35.932882
#13 2.033 https://nextjs.org/telemetry
2025-Jun-17 11:28:35.932882
#13 2.034
2025-Jun-17 11:28:35.932882
#13 2.126    ▲ Next.js 15.3.3
2025-Jun-17 11:28:35.932882
#13 2.127
2025-Jun-17 11:28:35.932882
#13 2.196    Creating an optimized production build ...
2025-Jun-17 11:28:35.932882
#13 17.88  ✓ Compiled successfully in 11.0s
2025-Jun-17 11:28:35.932882
#13 17.89    Linting and checking validity of types ...
2025-Jun-17 11:28:35.932882
#13 21.88
2025-Jun-17 11:28:35.932882
#13 21.88 Failed to compile.
2025-Jun-17 11:28:35.932882
#13 21.88
2025-Jun-17 11:28:35.932882
#13 21.88 ./src/app/cameras/[id]/page.tsx
2025-Jun-17 11:28:35.932882
#13 21.88 18:3  Error: 'Play' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
#13 21.88 29:9  Error: 'router' is assigned a value but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
#13 21.88 172:21  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 11:28:35.932882
#13 21.88 369:38  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
2025-Jun-17 11:28:35.932882
#13 21.88
2025-Jun-17 11:28:35.932882
#13 21.88 ./src/app/cameras/page.tsx
2025-Jun-17 11:28:35.932882
#13 21.88 146:27  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 11:28:35.932882
#13 21.88
2025-Jun-17 11:28:35.932882
#13 21.88 ./src/hooks/use-api.ts
2025-Jun-17 11:28:35.932882
#13 21.88 95:17  Error: '_' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
#13 21.88 95:20  Error: 'configId' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
#13 21.88
2025-Jun-17 11:28:35.932882
#13 21.88 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-17 11:28:35.932882
#13 ERROR: process "/bin/bash -ol pipefail -c npm run build" did not complete successfully: exit code: 1
2025-Jun-17 11:28:35.932882
------
2025-Jun-17 11:28:35.932882
> [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build:
2025-Jun-17 11:28:35.932882
21.88 369:38  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
2025-Jun-17 11:28:35.932882
21.88
2025-Jun-17 11:28:35.932882
21.88 ./src/app/cameras/page.tsx
2025-Jun-17 11:28:35.932882
21.88 146:27  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 11:28:35.932882
21.88
2025-Jun-17 11:28:35.932882
21.88 ./src/hooks/use-api.ts
2025-Jun-17 11:28:35.932882
21.88 95:17  Error: '_' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
21.88 95:20  Error: 'configId' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 11:28:35.932882
21.88
2025-Jun-17 11:28:35.932882
21.88 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-17 11:28:35.932882
------
2025-Jun-17 11:28:35.932882
Dockerfile:24
```


</details>

Shivansh1357 committed:
Enhance camera snapshot handling and improve API integration

- Replaced <img> tags with <Image> component for better performance and optimization in camera pages.
- Added width and height attributes to <Image> components to improve layout stability.
- Updated success callback in useDeleteDemographicsConfig to simplify the function.
- Removed unused router import in CameraDetailPage for cleaner code.

Shivansh1357 commented:
> @Shivansh1357 the latest build has failed again please refer to the logs below
> 
> Build logs

Resolved, Please Merge Again!

shafay07 commented:
The updated build is failing too please do a clean build locally to fix all the issues


<details>
<summary>
Build logs
</summary>

```
#13 [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build
2025-Jun-17 13:33:37.769542
#13 0.574 npm warn config production Use `--omit=dev` instead.
2025-Jun-17 13:33:37.769542
#13 0.608
2025-Jun-17 13:33:37.769542
#13 0.608 > my-app@0.1.0 build
2025-Jun-17 13:33:37.769542
#13 0.608 > next build
2025-Jun-17 13:33:37.769542
#13 0.608
2025-Jun-17 13:33:37.769542
#13 1.943    ▲ Next.js 15.3.3
2025-Jun-17 13:33:37.769542
#13 1.944
2025-Jun-17 13:33:37.769542
#13 2.009    Creating an optimized production build ...
2025-Jun-17 13:33:37.769542
#13 17.72  ✓ Compiled successfully in 11.0s
2025-Jun-17 13:33:37.769542
#13 17.73    Linting and checking validity of types ...
2025-Jun-17 13:33:37.769542
#13 21.69
2025-Jun-17 13:33:37.769542
#13 21.69 Failed to compile.
2025-Jun-17 13:33:37.769542
#13 21.69
2025-Jun-17 13:33:37.769542
#13 21.69 ./src/app/cameras/[id]/page.tsx
2025-Jun-17 13:33:37.769542
#13 21.69 18:3  Error: 'Play' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
#13 21.69 29:9  Error: 'router' is assigned a value but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
#13 21.69 172:21  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 13:33:37.769542
#13 21.69 369:38  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
2025-Jun-17 13:33:37.769542
#13 21.69
2025-Jun-17 13:33:37.769542
#13 21.69 ./src/app/cameras/page.tsx
2025-Jun-17 13:33:37.769542
#13 21.69 146:27  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 13:33:37.769542
#13 21.69
2025-Jun-17 13:33:37.769542
#13 21.69 ./src/hooks/use-api.ts
2025-Jun-17 13:33:37.769542
#13 21.69 95:17  Error: '_' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
#13 21.69 95:20  Error: 'configId' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
#13 21.69
2025-Jun-17 13:33:37.769542
#13 21.69 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-17 13:33:37.769542
#13 ERROR: process "/bin/bash -ol pipefail -c npm run build" did not complete successfully: exit code: 1
2025-Jun-17 13:33:37.769542
------
2025-Jun-17 13:33:37.769542
> [stage-0  9/11] RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build:
2025-Jun-17 13:33:37.769542
21.69 369:38  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
2025-Jun-17 13:33:37.769542
21.69
2025-Jun-17 13:33:37.769542
21.69 ./src/app/cameras/page.tsx
2025-Jun-17 13:33:37.769542
21.69 146:27  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
2025-Jun-17 13:33:37.769542
21.69
2025-Jun-17 13:33:37.769542
21.69 ./src/hooks/use-api.ts
2025-Jun-17 13:33:37.769542
21.69 95:17  Error: '_' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
21.69 95:20  Error: 'configId' is defined but never used.  @typescript-eslint/no-unused-vars
2025-Jun-17 13:33:37.769542
21.69
2025-Jun-17 13:33:37.769542
21.69 info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/app/api-reference/config/eslint#disabling-rules
2025-Jun-17 13:33:37.769542
------
2025-Jun-17 13:33:37.769542
Dockerfile:24
2025-Jun-17 13:33:37.769542
--------------------
2025-Jun-17 13:33:37.769542
22 |     # build phase
2025-Jun-17 13:33:37.769542
23 |     COPY . /app/.
2025-Jun-17 13:33:37.769542
24 | >>> RUN --mount=type=cache,id=jo0c4w4cc400884sos4488ww-next/cache,target=/app/.next/cache --mount=type=cache,id=jo0c4w4cc400884sos4488ww-node_modules/cache,target=/app/node_modules/.cache npm run build
2025-Jun-17 13:33:37.769542
25 |
2025-Jun-17 13:33:37.769542
26 |
2025-Jun-17 13:33:37.769542
--------------------
2025-Jun-17 13:33:37.769542
ERROR: failed to solve: process "/bin/bash -ol pipefail -c npm run build" did not complete successfully: exit code: 1
2025-Jun-17 13:33:37.769542
exit status 1



```


</details>

Shivansh1357 commented:
> The updated build is failing too please do a clean build locally to fix all the issues
> 
> Build logs


sorry i forgot to push the code after committing it, can you merge now?

Shivansh1357 committed:
Enhance UI components and configuration settings

- Updated global CSS variables for improved color contrast and dark mode support.
- Modified camera page styles for better readability and consistency.
- Enhanced button and input components with updated styles and hover effects.
- Improved navbar appearance with shadow effects for better visual hierarchy.
- Adjusted card component styles for a more modern look and feel.

Shivansh1357 commented:
> The updated build is failing too please do a clean build locally to fix all the issues
> 
> Build logs

can you merge Again

Shivansh1357 committed:
Update text color for improved readability across multiple pages

- Changed text color from gray-600 to gray-800 in various components to enhance contrast and readability.
- Updated error messages and informational texts in camera, analytics, and demographics configuration pages for consistency.

Shivansh1357 commented:
> The updated build is failing too please do a clean build locally to fix all the issues
> 
> Build logs

Hi, can you merge and check for other pages as well?

Shivansh1357 committed:
Enhance UI and functionality across multiple components

- Added new Radix UI components for improved user experience, including avatar, dialog, dropdown menu, and tabs.
- Updated global CSS variables for better color contrast and dark mode support.
- Enhanced button, card, and input components with updated styles and hover effects.
- Improved camera and analytics pages with better layout and responsiveness.
- Integrated theme provider for consistent theming across the application.
- Updated package dependencies in package.json and package-lock.json for better performance and compatibility.

Shivansh1357 commented:
> The updated build is failing too please do a clean build locally to fix all the issues
> 
> Build logs

Hi Shafay,

Apologies for the delay. I've completely redesigned the UI using ShadCN's modern component library to address the styling issues. For your convenience, I’ve also added dark and light mode themes for better user experience.

Please have a look and let me know your thoughts.
